### PR TITLE
Update information regarding the 'state' parameter for OAuth.

### PR DESCRIPTION
**Pull request by aw-asana**
Screen Shot 2022-03-22 at 2 41 44 PM
Screen Shot 2022-03-22 at 2 42 11 PM

### CODE REVIEW

**Review by jv-asana — APPROVED** [review]
LGTM

File: source/includes/markdown/_07-auth.html.md [hunk 1]
```diff
@@ -146,8 +146,11 @@ When using the `response_type=code`, your app will receive the following paramet
 | **code** | If response_type=code in the request, this is the code your app can exchange for a token |
 | **state** | The state parameter that was sent with the authorizing request |
 
-You should check that the state is the same in this response as it was in the request. You can read more about the `state`
-parameter [here](https://auth0.com/docs/protocols/oauth2/oauth-state).
+The `state` parameter is necessary to prevent CSRF attacks. As such, **you should check that the `state` is the same in this response as it was in the request.**
```
Comment (jadengeller-asana) [inline source/includes/markdown/_07-auth.html.md:149]: nit: I wonder if `must` is better than `should` here.

File: source/includes/markdown/_07-auth.html.md [hunk 2]
```diff
@@ -146,8 +146,11 @@ When using the `response_type=code`, your app will receive the following paramet
...
+The `state` parameter is necessary to prevent CSRF attacks. As such, **you should check that the `state` is the same in this response as it was in the request.** 
+
+The `state` parameter must contain an unguessable value tied to the user's session, which can be the hash of something tied to their session when the OAuth flow is first initiated (e.g., their session cookie). This value is then passed back and forth between the client application and the OAuth service as a form of CSRF token for the client application. If the `state` parameter is _not_ used, or _not_ tied to the user's session, then attackers can initiate an OAuth flow themselves before tricking a user's browser into completing it. This is similar to a traditional CSRF attack and can also lead to victims binding their accounts to an attacker account without their knowledge. See the [OAuth 2.0 Security best practices](https://tools.ietf.org/id/draft-ietf-oauth-security-topics-13.html) for more detailed information.
+
+For more general information about preventing CSRF attacks with the `state` parameter, feel free to visit [this article](https://auth0.com/docs/secure/attack-protection/state-parameters).
```
Comment (jadengeller-asana) [inline source/includes/markdown/_07-auth.html.md:153]: nit: I'm wondering how this link is different than the link in the previous paragraph. If we want to provide both resources, should we do so in the same paragraph? Or is just one of these sufficient? It feels like we're repeating ourselves a bit

File: source/includes/markdown/_07-auth.html.md [hunk 3]
```diff
@@ -146,8 +146,11 @@ When using the `response_type=code`, your app will receive the following paramet
...
 
-You should check that the state is the same in this response as it was in the request. You can read more about the `state`
-parameter [here](https://auth0.com/docs/protocols/oauth2/oauth-state).
+The `state` parameter is necessary to prevent CSRF attacks. As such, **you should check that the `state` is the same in this response as it was in the request.** 
+
+The `state` parameter must contain an unguessable value tied to the user's session, which can be the hash of something tied to their session when the OAuth flow is first initiated (e.g., their session cookie). This value is then passed back and forth between the client application and the OAuth service as a form of CSRF token for the client application. If the `state` parameter is _not_ used, or _not_ tied to the user's session, then attackers can initiate an OAuth flow themselves before tricking a user's browser into completing it. This is similar to a traditional CSRF attack and can also lead to victims binding their accounts to an attacker account without their knowledge. See the [OAuth 2.0 Security best practices](https://tools.ietf.org/id/draft-ietf-oauth-security-topics-13.html) for more detailed information.
```
Comment (jadengeller-asana) [inline source/includes/markdown/_07-auth.html.md:151]: nit: This paragraph is a bit long, so I wonder how many developers might skip over it. Could it be worth reformatting this into bullet points to make the requirements more skimmable?

File: source/includes/markdown/_07-auth.html.md (hunk 1 above)
Comment (aw-asana) [reply source/includes/markdown/_07-auth.html.md:149]: `must`'d!

File: source/includes/markdown/_07-auth.html.md (hunk 2 above)
Comment (aw-asana) [reply source/includes/markdown/_07-auth.html.md:153]: I added an Additional Resources section

File: source/includes/markdown/_07-auth.html.md (hunk 3 above)
Comment (aw-asana) [reply source/includes/markdown/_07-auth.html.md:151]: Good call - I made the separations more clearer now

**Comment by aw-asana** [comment]
Screen Shot 2022-03-23 at 2 38 50 PM

**Review by jadengeller-asana — APPROVED** [review]

**Pull request merged**